controllers/openstackdataplanerole_controller.go

```diff
@@ -294,6 +305,84 @@ func (r *OpenStackDataPlaneRoleReconciler) ReconcileBaremetalSet(ctx context.Con
 	return nil, nil
 }
 
+// ServiceYAML struct for service YAML unmarshalling
+type ServiceYAML struct {
+	Kind     string
+	Metadata yaml.Node
+	Spec     yaml.Node
+}
+
+// EnsureServices - ensure the OpenStackDataPlaneServices exist
+func (r *OpenStackDataPlaneRoleReconciler) EnsureServices(ctx context.Context, helper *helper.Helper, instance *dataplanev1beta1.OpenStackDataPlaneRole) error {
+	servicesPath, found := os.LookupEnv("OPERATOR_SERVICES")
+	if !found {
+		servicesPath = "config/services"
+		os.Setenv("OPERATOR_SERVICES", servicesPath)
+		util.LogForObject(
+			helper, "OPERATOR_SERVICES not set in env when reconciling ", instance,
+			"defaulting to ", servicesPath)
+	}
+
+	r.Log.Info("Ensuring services", "servicesPath", servicesPath)
+	services, err := os.ReadDir(servicesPath)
+	if err != nil {
+		return err
+	}
+
+	for _, service := range services {
```

Comment:
It would be better to reconcile only the services that the role uses, have in the spec an not all.

Reply:
Ok, I made that change.